HeliPort/Appearance/StatusMenu.swift

```diff
@@ -435,7 +435,11 @@ final class StatusMenu: NSMenu, NSMenuDelegate {
             }
         case .turnWiFiOn:
             power_on()
+            NetworkManager.scanSavedNetworks()
         case .turnWiFiOff:
+            if status == ITL80211_S_RUN {
+                disassociateSSID(disconnectItem)
+            }
```

Comment:
When connected, it will disconnect from the current network before turning off the Wi-Fi, thus preventing it from connecting to a network with auto-join disabled when Wi-Fi is turned on (if the previously connected network has auto-join disabled).

Reply:
With the changes made to `.turnWiFiOn` and `.turnWiFiOff`, it behaves the same as Apple's implementation, at least in more recent versions of macOS, as I'm not sure if the same happens in earlier versions.